# Patch for compiling codeblocks with gcc-4.1
# Priyank <priyankmg@gmail.com>

--- src/sdk/managedthread.h	2006-05-30 15:49:01.000000000 +0530
+++ src/sdk/managedthread.h	2006-05-30 15:49:16.000000000 +0530
@@ -11,8 +11,8 @@
 class ManagedThread : public wxThread
 {
 public:
-    ManagedThread::ManagedThread(bool* abortflag  = 0L);
-    virtual ManagedThread::~ManagedThread();
+    ManagedThread(bool* abortflag  = 0L);
+    virtual ~ManagedThread();
     static unsigned long count_running();
     static unsigned long count_threads();
     static void abort_all(); // Warning! Once set, can't be reset!

--- src/sdk/editormanager.h	2006-05-30 15:56:16.000000000 +0530
+++ src/sdk/editormanager.h	2006-05-30 15:57:07.000000000 +0530
@@ -126,7 +126,7 @@
 
         /** Builds Opened Files tree in the Projects tab
           */
-        wxTreeCtrl *EditorManager::GetTree();
+        wxTreeCtrl* GetTree();
         wxTreeItemId FindTreeFile(const wxString& filename);
         wxString GetTreeItemFilename(wxTreeItemId item);
         void BuildOpenedFilesTree(wxWindow* parent);

--- src/sdk/xtra_classes.h	2006-05-30 15:51:27.000000000 +0530
+++ src/sdk/xtra_classes.h	2006-05-30 15:52:24.000000000 +0530
@@ -18,8 +18,8 @@
 class wxSplitPanel : public wxPanel
 {
 public:
-    wxSplitPanel::wxSplitPanel() { /*nothing to init, really */ }
-    wxSplitPanel::wxSplitPanel(wxWindow* parent, wxWindowID id = -1,
+    wxSplitPanel() { /*nothing to init, really */ }
+    wxSplitPanel(wxWindow* parent, wxWindowID id = -1,
                                const wxPoint& pos = wxDefaultPosition, const wxSize& size = wxDefaultSize,
                                long style = wxTAB_TRAVERSAL, const wxString& name = _T("splitpanel"),const wxString configname = wxEmptyString,int defaultsashposition = 150)
     {
@@ -33,9 +33,9 @@
       * If only one is found, the panel is not split.
       *
       */
-    void wxSplitPanel::RefreshSplitter(int idtop,int idbottom);
+    void RefreshSplitter(int idtop,int idbottom);
 
-    virtual wxSplitPanel::~wxSplitPanel();
+    virtual ~wxSplitPanel();
     wxSplitterWindow* GetSplitter() { return (this) ? m_splitter : 0L; }
     void SetConfigEntryForSplitter(const wxString& splitterconfig){ m_SplitterConfig = splitterconfig; }
 protected:

--- src/sdk/cbthreadpool.cpp		2006-05-30 16:10:00.000000000 +0530
+++ src/sdk/cbthreadpool.cpp		2006-05-30 16:10:57.000000000 +0530
@@ -250,7 +250,7 @@
 #if wxCHECK_VERSION(2,6,0)
     wxMilliSleep(1);
 #else
-    wxUSleep(1);
+    wxUsleep(1);
 #endif
 
     wxLogNull logNo;
@@ -265,7 +265,7 @@
 #if wxCHECK_VERSION(2,6,0)
             wxMilliSleep(1);
 #else
-            wxUSleep(1);
+            wxUsleep(1);
 #endif
             if(++count > 10)
                 break;
